specs/CommandBar/new-command-bar-features.md

```diff
@@ -0,0 +1,349 @@
+If a child is selected and the `MenuFlyout` is open, both the parent and child will show the selection indicator (similar to a hierarchical `NavigationView`).
+
+These are the three main visual states for this new API, using the same Xaml Controls Gallery sample that was previously shown:
+
+*Child item selected, flyout open*
+
+![Child item selected, flyout open](images/commandbar2.png)
+
+*Child item selected, flyout closed*
+
+![Child item selected, flyout closed](images/commandbar3.png)
+
+*Other top-level item selected*
+
+![Top-level item selected](images/commandbar1.PNG)	 
+
+For example:
+
+```xml
+<CommandBar DefaultLabelPosition="Right" Grid.Row="1" Margin="50">
+    <AppBarToggleButton Icon="Shuffle" Label="Shuffle" />
+    <AppBarToggleButton Icon="RepeatAll" Label="Repeat" />
+    <AppBarSeparator/>
+    <AppBarButton Icon="Back" />
+    <AppBarButton Icon="Stop" />
+    <AppBarButton Icon="Go" Label="Sort by">
+        <AppBarButton.Flyout>
+            <MenuFlyout>
+                <RadioMenuFlyoutItem Text="Name" GroupName="SortOption"/>
+                <RadioMenuFlyoutItem Text="Date" GroupName="SortOption" 
+                                     IsChecked="True"/>
+                <RadioMenuFlyoutItem Text="Size" GroupName="SortOption"/>
+                <RadioMenuFlyoutSubItem Text="Other">
+                    <RadioMenuFlyoutItem Text="Album name" GroupName="SortOption"/>
+                    <RadioMenuFlyoutItem Text="Artist name" GroupName="SortOption"/>
+                    <RadioMenuFlyoutItem Text="Genre" GroupName="SortOption"/>
+                </RadioMenuFlyoutSubItem>
+            </MenuFlyout>
+        </AppBarButton.Flyout>
+    </AppBarButton>
+
+    <AppBarSeparator/>
+    <AppBarButton Icon="Play" Label="Play" />
+    <AppBarButton Icon="Forward" Label="Forward" />
+
+</CommandBar>
+```
+
+
+### Feature #3. AlwaysExpanded API
+
+This will be a new API added to `CommandBarFlyout` that will give the developer the ability to keep the `CommandBarFlyout` in its expanded mode at all times.
+
+When the AlwaysExpanded property is set to true, the [...] button will not appear, 
+and the user will not be able to collapse the `CommandBarFlyout`. 
+Once the [...] button is collapsed, other `AppBarButtons`/`CommandBarFlyout` items should be able to take its space. 
+
+This property will only have an effect if the `CommandBarFlyout` has secondary commands. 
+If there are no secondary commands, the `CommandBarFlyout` will always be in collapsed mode (hence why there’s no `AlwaysCollapsed` property). 
+
+`CommandBarFlyout` can still be collapsed/expanded by the developer programmatically even when this property is set to true. When an item in the primary commands list is invoked, the CommandBarFlyout should stay open and expanded. If an item in the secondary commands list is invoked, the CommandBarFlyout should close completely. 
```

Comment:
> ... the CommandBarFlyout should stay open and expanded.

I think this should be "should stay open and not change its collapsed/expanded state." Because picking a primary command from a collapsed CommandBarFlyout presumably does not expand it.

Reply:
Recommend